>From d4f07d73ea4795c66857f8180dd5d822ec48fc47 Mon Sep 17 00:00:00 2001
From: Isaac Jurado <diptongo@gmail.com>
Date: Sat, 30 Jan 2016 18:26:53 +0100
Subject: [PATCH] dh_auto_configure: Expand ${prefix} uses

Some configure scripts look like generated by autoconf (e.g. MakeKit) but don't
behave exactly the same when expanding the install paths.  For example, when
dh_auto_configure uses "--includedir=${prefix}/include", the "${prefix}" part is
NOT always re-evaluated and, thus, expanded to "/usr".
---
 Debian/Debhelper/Buildsystem/autoconf.pm | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/Debian/Debhelper/Buildsystem/autoconf.pm b/Debian/Debhelper/Buildsystem/autoconf.pm
index 62ff8b3..1d61c0b 100644
--- a/Debian/Debhelper/Buildsystem/autoconf.pm
+++ b/Debian/Debhelper/Buildsystem/autoconf.pm
@@ -31,11 +31,12 @@ sub configure {
 
 	# Standard set of options for configure.
 	my @opts;
+	my $prefix="/usr";
 	push @opts, "--build=" . dpkg_architecture_value("DEB_BUILD_GNU_TYPE");
-	push @opts, "--prefix=/usr";
-	push @opts, "--includedir=\${prefix}/include";
-	push @opts, "--mandir=\${prefix}/share/man";
-	push @opts, "--infodir=\${prefix}/share/info";
+	push @opts, "--prefix=$prefix";
+	push @opts, "--includedir=$prefix/include";
+	push @opts, "--mandir=$prefix/share/man";
+	push @opts, "--infodir=$prefix/share/info";
 	push @opts, "--sysconfdir=/etc";
 	push @opts, "--localstatedir=/var";
 	if (defined $ENV{DH_QUIET} && $ENV{DH_QUIET} ne "") {
@@ -46,15 +47,15 @@ sub configure {
 	my $multiarch=dpkg_architecture_value("DEB_HOST_MULTIARCH");
 	if (! compat(8)) {
 	       if (defined $multiarch) {
-			push @opts, "--libdir=\${prefix}/lib/$multiarch";
-			push @opts, "--libexecdir=\${prefix}/lib/$multiarch";
+			push @opts, "--libdir=$prefix/lib/$multiarch";
+			push @opts, "--libexecdir=$prefix/lib/$multiarch";
 		}
 		else {
-			push @opts, "--libexecdir=\${prefix}/lib";
+			push @opts, "--libexecdir=$prefix/lib";
 		}
 	}
 	else {
-		push @opts, "--libexecdir=\${prefix}/lib/" . sourcepackage();
+		push @opts, "--libexecdir=$prefix/lib/" . sourcepackage();
 	}
 	push @opts, "--disable-maintainer-mode";
 	push @opts, "--disable-dependency-tracking";
-- 
2.7.0

